src/index.test.jsx

```diff
@@ -230,4 +230,163 @@ describe('web components', () => {
 		});
 		assert.equal(getShadowHTML(), '<p>Active theme: sunny</p>');
 	});
+
+	function Thing() {
+		return <span>Hello world!</span>;
+	}
+
+	const styleElem = document.createElement('style');
+	styleElem.innerHTML = 'span { color: red; }';
+	document.head.appendChild(styleElem);
+
+	registerElement(Thing, 'x-thing', undefined, {
+		shadow: true,
+		injectGlobalStyles: true,
+	});
+
+	describe('Global style injections from document.head', () => {
+		it('injects style-tags', () => {
+			const el = document.createElement('x-thing');
+
+			root.appendChild(el);
+
+			assert.equal(
+				document.querySelector('x-thing').shadowRoot.innerHTML,
+				'<span>Hello world!</span><style>span { color: red; }</style>'
+			);
+
+			const computedStyle = window
+				.getComputedStyle(
+					document.querySelector('x-thing').shadowRoot.querySelector('span'),
+					null
+				)
+				.getPropertyValue('color');
+
+			assert.equal(computedStyle, 'rgb(255, 0, 0)');
+
+			// assert.equal(
+			// 	root.innerHTML,
+			// 	'<x-thing><span>Hello world!</span></x-thing>'
+			// );
+			styleElem.parentElement.removeChild(styleElem);
+		});
+
+		it('injects link-tags of rel="stylesheet"', async () => {
+			const blob = new Blob([], { type: 'text/css' });
+
+			let linkElementLoaded;
+
+			let deferred;
+			let promise = new Promise((resolve) => {
+				deferred = resolve;
+			});
+
+			const linkElem = document.createElement('link');
+			linkElem.rel = 'stylesheet';
+			linkElem.href = window.URL.createObjectURL(blob);
+			linkElem.onload = () => {
+				linkElementLoaded = true;
+				deferred();
+			};
+
+			document.head.appendChild(linkElem);
+
+			const el = document.createElement('x-thing');
+
+			root.appendChild(el);
+
+			assert.match(
+				document.querySelector('x-thing').shadowRoot.innerHTML,
+				new RegExp(
+					`<span>Hello world!</span><link rel="stylesheet" href="blob:http://.*?/[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}">`
```

Comment:
The tests are failing because the port is picked randomly by the test runner, based on what's the next free one on your machine. 

Since we construct the `link`-tag ourselves, we already have the URL. Therefor we can replace the regex with simple string concatenation:

```js
assert.equal(
  document.querySelector('x-thing').shadowRoot.innerHTML,
  `<span>Hello world!</span><link rel="stylesheet" href="${linkElem.href}">`
);
```

That way we're not making ourselves dependent of the test runner's implementation on how it constructs URLs :+1:

Reply:
:+1: